src/server/javascript.ts

```diff
@@ -174,28 +173,25 @@ export async function evaluate(context: ExecutionContext, returnByValue: boolean
   return evaluateExpression(context, returnByValue, String(pageFunction), typeof pageFunction === 'function', ...args);
 }
 
-export async function evaluateExpression(context: ExecutionContext, returnByValue: boolean, expression: string, isFunction: boolean, ...args: any[]): Promise<any> {
+export async function evaluateExpression(context: ExecutionContext, returnByValue: boolean, expression: string, isFunction: boolean | undefined, ...args: any[]): Promise<any> {
   const utilityScript = await context.utilityScript();
-  if (!isFunction) {
-    const script = `(utilityScript, ...args) => utilityScript.evaluate(...args)`;
-    return context._delegate.evaluateWithArguments(script, returnByValue, utilityScript, [returnByValue, expression], []);
-  }
 
-  let functionText = expression;
-  try {
-    new Function('(' + functionText + ')');
-  } catch (e1) {
-    // This means we might have a function shorthand. Try another
-    // time prefixing 'function '.
-    if (functionText.startsWith('async '))
-      functionText = 'async function ' + functionText.substring('async '.length);
-    else
-      functionText = 'function ' + functionText;
+  if (isFunction) {
     try {
-      new Function('(' + functionText  + ')');
-    } catch (e2) {
-      // We tried hard to serialize, but there's a weird beast here.
-      throw new Error('Passed function is not well-serializable!');
+      new Function('(' + expression + ')');
+    } catch (e1) {
+      // This means we might have a function shorthand. Try another
+      // time prefixing 'function '.
+      if (expression.startsWith('async '))
+        expression = 'async function ' + expression.substring('async '.length);
```

Comment:
I'd move this to the js client where we now if it's a function or not and just need to update the serialization logic to replace the prefix if needed. Could be done separately.

Reply:
I need to patch 10 call sites for that :/